usermods/Ble2Json_v2/usermod_v2_ble2json.h

```diff
@@ -0,0 +1,381 @@
+    if(cmpl.success){
+      Serial.println("   - SecurityCallback - Authentication Success");       
+    }else{
+      Serial.printf_P(PSTR("%u\n"),cmpl.fail_reason);
+      Serial.println("   - SecurityCallback - Authentication Failure* ");
+      pServer->removePeerDevice(pServer->getConnId(), true);
+    }
+    BLEDevice::startAdvertising();
+  }
+};
+
+void bleSecurity(){
+  esp_ble_auth_req_t auth_req = ESP_LE_AUTH_REQ_SC_BOND;
+  esp_ble_io_cap_t iocap = ESP_IO_CAP_OUT;          
+  uint8_t key_size = 16;     
+  uint8_t init_key = ESP_BLE_ENC_KEY_MASK | ESP_BLE_ID_KEY_MASK;
+  uint8_t rsp_key = ESP_BLE_ENC_KEY_MASK | ESP_BLE_ID_KEY_MASK;
+  uint32_t passkey = PASSKEY;
+  uint8_t auth_option = ESP_BLE_ONLY_ACCEPT_SPECIFIED_AUTH_DISABLE;
+  esp_ble_gap_set_security_param(ESP_BLE_SM_SET_STATIC_PASSKEY, &passkey, sizeof(uint32_t));
+  esp_ble_gap_set_security_param(ESP_BLE_SM_AUTHEN_REQ_MODE, &auth_req, sizeof(uint8_t));
+  esp_ble_gap_set_security_param(ESP_BLE_SM_IOCAP_MODE, &iocap, sizeof(uint8_t));
+  esp_ble_gap_set_security_param(ESP_BLE_SM_MAX_KEY_SIZE, &key_size, sizeof(uint8_t));
+  esp_ble_gap_set_security_param(ESP_BLE_SM_ONLY_ACCEPT_SPECIFIED_SEC_AUTH, &auth_option, sizeof(uint8_t));
+  esp_ble_gap_set_security_param(ESP_BLE_SM_SET_INIT_KEY, &init_key, sizeof(uint8_t));
+  esp_ble_gap_set_security_param(ESP_BLE_SM_SET_RSP_KEY, &rsp_key, sizeof(uint8_t));
+}
+
+void bleInit(BLECharacteristicCallbacks *charCallbackHandler){
+  Serial.println("bleInit");
+  BLEDevice::init("ble2json for wled");
+  BLEDevice::setEncryptionLevel(ESP_BLE_SEC_ENCRYPT);
+  BLEDevice::setSecurityCallbacks(new SecurityCallback());
+
+  pServer = BLEDevice::createServer();
+  pServer->setCallbacks(new ServerCallback());
+
+  BLEService *pService = pServer->createService(SERVICE_UUID);
+  pCharacteristic = pService->createCharacteristic(
+                      CHARACTERISTIC_UUID,
+                      BLECharacteristic::PROPERTY_READ   |
+                      BLECharacteristic::PROPERTY_WRITE  |
+                      BLECharacteristic::PROPERTY_NOTIFY 
+                    );
+
+  pCharacteristic->setAccessPermissions(ESP_GATT_PERM_READ_ENCRYPTED | ESP_GATT_PERM_WRITE_ENCRYPTED);
+  pCharacteristic->setCallbacks(charCallbackHandler);
+
+  pService->start();
+
+  BLEAdvertising *pAdvertising = BLEDevice::getAdvertising();
+  pAdvertising->addServiceUUID(SERVICE_UUID);
+  pAdvertising->setScanResponse(false);
+  pAdvertising->setMinPreferred(0x0);
+  BLEDevice::startAdvertising();
+
+  bleSecurity();
+}
+
+class Ble2JsonCallbacks {
+  public:
+    virtual void turnOffBle() = 0;
+};
+
+class MyCallbackHandler: public BLECharacteristicCallbacks {
+  private: 
+    Ble2JsonCallbacks *userMod;
+
+  public: 
+    MyCallbackHandler(Ble2JsonCallbacks *_userMod) {
+      userMod = _userMod;
+    }
+
+	// void onRead(BLECharacteristic *pCharacteristic) {
+	// }
+
+	void onWrite(BLECharacteristic* pCharacteristic) {
```

Comment:
Indentation.

Reply:
I think i've addressed all the formatting issues... I installed prettier and have auto-format on... then, when I edit an existing wled file, I just edit in `vi` so as not to invoke the auto-formatting